### PR TITLE
Automated splitting of CHANGELOG.md

### PR DESCRIPTION
**Pull request by Shubhabrata08**
This PR automates the updation of changelog and splitting the entire changelog for better indexing at docsite.
It runs weekly once, fetches the latest changelog, updates it and creates a PR for the changes.
Here is a demo changelog update PR :
https://github.com/Shubhabrata08/fluentui-charting-contrib/pull/2
Here is a demo workflow run:
https://github.com/Shubhabrata08/fluentui-charting-contrib/actions/runs/8033075840
Fixes: #58

### CODE REVIEW

File: apps/docsite/src/scripts/splitChangelog.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        if (versionSplit[0] === "5") {
+          const minorVersion = versionSplit[0] + "." + versionSplit[1];
+          if (currentVersion != minorVersion) {
+            currentVersion = minorVersion;
+            versionToLineMapping.push({
+              version: currentVersion,
+              startLine: index,
+            });
+          }
+        } else if (versionSplit[0] === "4") {
```
Comment (AtishayMsft) [inline apps/docsite/src/scripts/splitChangelog.ts:44]: why is the else if statement needed

Comment (Shubhabrata08) [reply apps/docsite/src/scripts/splitChangelog.ts:44]: That's for storing the line after which  version 5.0.0 ends

**Comment by AtishayMsft** [comment]
Can we have the version.md files in reverse order with latest version at the top.

**Comment by Shubhabrata08** [comment]
image
We can generate an automated JSON and use it to create our custom order and add it to sidebar.ts
image

**Comment by Shubhabrata08** [comment]
Shall I add this to the PR?

**Comment by AtishayMsft** [comment]
> Shall I add this to the PR?

yes we can.

File: apps/docsite/sidebars.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
-import type {SidebarsConfig} from '@docusaurus/plugin-content-docs';
-
+import type { SidebarsConfig } from "@docusaurus/plugin-content-docs";
+import changelog from "../../docs/changelogSplits/changelogOrder.json";
```
Comment (AtishayMsft) [inline apps/docsite/sidebars.ts:2]: this will break till a seed change log order file is created

Comment (Shubhabrata08) [reply apps/docsite/sidebars.ts:2]: We can either merge and run the workfloe to generate the seed or manually push one

Comment (AtishayMsft) [reply apps/docsite/sidebars.ts:2]: lets try running the workflow after merging

**Review by AtishayMsft — APPROVED** [review]

**Pull request merged**